File: Exercise/Very Hard/10. Casino Security/Code/Casino Security.py
def casino_security(s):
    thief_pos = s.find('T')
    money_pos = s.find('$')
    guard_pos = s.find('G')

    if thief_pos == -1 or money_pos == -1:
        return "Safe"

    if (thief_pos < money_pos < guard_pos) or (guard_pos < money_pos < thief_pos):
        return "Safe"

    return "ALARM!"

def main():
    input_str = input()
    result = casino_security(input_str)
    print(result)

if __name__ == "__main__":
    main()

File: Exercise/Very Hard/10. Casino Security/Code/Casino Security.java
import java.util.Scanner;

public class CasinoSecurity {

    static String casinoSecurity(String s) {
        int thiefPos = s.indexOf('T');
        int moneyPos = s.indexOf('$');
        int guardPos = s.indexOf('G');

        if (thiefPos == -1 || moneyPos == -1) {
            return "Safe";
        }

        if ((thiefPos < moneyPos && moneyPos < guardPos) ||
            (guardPos < moneyPos && moneyPos < thiefPos)) {
            return "Safe";
        }

        return "ALARM!";
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();

        String result = casinoSecurity(input);
        System.out.println(result);
    }
}




